src/main/java/com/limechain/sync/warpsync/state/VerifyJustificationState.java

```diff
@@ -0,0 +1,127 @@
+package com.limechain.sync.warpsync.state;
+
+import com.limechain.chain.lightsyncstate.Authority;
+import com.limechain.network.protocol.warp.dto.ConsensusEngine;
+import com.limechain.network.protocol.warp.dto.HeaderDigest;
+import com.limechain.network.protocol.warp.dto.WarpSyncFragment;
+import com.limechain.sync.warpsync.dto.AuthoritySetChange;
+import com.limechain.sync.warpsync.dto.GrandpaMessageType;
+import com.limechain.sync.warpsync.WarpSyncMachine;
+import com.limechain.sync.warpsync.scale.ForcedChangeReader;
+import com.limechain.sync.warpsync.scale.ScheduledChangeReader;
+import io.emeraldpay.polkaj.scale.ScaleCodecReader;
+import lombok.extern.java.Log;
+import org.javatuples.Pair;
+
+import java.math.BigInteger;
+import java.util.Queue;
+import java.util.logging.Level;
+
+// VerifyJustificationState is going to be instantiated a lot of times
+// Maybe we can make it a singleton in order to reduce performance overhead?
+@Log
+public class VerifyJustificationState implements WarpSyncState {
+    private Exception error;
+
+    @Override
+    public void next(WarpSyncMachine sync) {
+        if (this.error != null) {
+            // Not sure what state we should transition to here.
+            sync.setState(new FinishedState());
+            return;
+        }
+
+        if (!sync.getFragmentsQueue().isEmpty()) {
+            sync.setState(new VerifyJustificationState());
+        } else if (sync.isFinished()) {
+            sync.setState(new RuntimeDownloadState());
+        } else {
+            sync.setState(new RequestFragmentsState(sync.getLastFinalizedBlockHash()));
+        }
+    }
+
+    @Override
+    public void handle(WarpSyncMachine sync) {
+        try {
+            handleScheduledEvents(sync);
+
+            WarpSyncFragment fragment = sync.getFragmentsQueue().poll();
+            log.log(Level.INFO, "Verifying justification...");
+            // TODO: Throw error if not verified!
+            boolean verified = fragment.getJustification().verify(sync.getAuthoritySet(), sync.getSetId());
+            if (!verified) {
+                throw new Exception("Justification could not be verified.");
+            }
```

Comment:
Isn't this TODO resolved? Also It's always better to throw unchecked exceptions, the mother of all unchecked exceptions is `RuntimeException`.